src/librustc_mir/borrow_check.rs

```diff
@@ -744,27 +751,29 @@ impl<'cx, 'gcx, 'tcx> MirBorrowckCtxt<'cx, 'gcx, 'tcx> {
 
     /// Returns whether a borrow of this place is invalidated when the function
     /// exits
-    fn place_is_invalidated_at_exit(&self, place: &Place<'tcx>) -> bool {
+    fn place_is_invalidated_at_exit(&mut self, place: &Place<'tcx>) -> bool {
         debug!("place_is_invalidated_at_exit({:?})", place);
         let root_place = self.prefixes(place, PrefixSet::All).last().unwrap();
 
         // FIXME(nll-rfc#40): do more precise destructor tracking here. For now
         // we just know that all locals are dropped at function exit (otherwise
         // we'll have a memory leak) and assume that all statics have a destructor.
-        let (might_be_alive, will_be_dropped) = match root_place {
+        //
+        // FIXME: allow thread-locals to borrow other thread locals?x
```

Comment:
Can you say a bit more about this? It seems like you have disabled these checks in statics/constants, right? Is that what you meant?

Reply:
Oh, never mind, that flag was for whether *locals* are considered dropped, right...